var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -38,67 +45,184 @@ class Babelstream(CMakePackage, CudaPackage, ROCmPackage):
     variant("ocl", default=False, description="Enable OpenCL support")
     variant("tbb", default=False, description="Enable TBB support")
     variant("acc", default=False, description="Enable OpenACC support")
+    variant("hip", default=False, description="Enable HIP support")
     variant("thrust", default=False, description="Enable THRUST support")
     variant("raja", default=False, description="Enable RAJA support")
-    variant("stddata", default=False, description="Enable STD-data support")
-    variant("stdindices", default=False, description="Enable STD-indices support")
-    variant("stdranges", default=False, description="Enable STD-ranges support")
+    variant("std", default=False, description="Enable STD support")
 
     # Some models need to have the programming model abstraction downloaded -
     # this variant enables a path to be provided.
     variant("dir", values=str, default="none", description="Enable Directory support")
+    variant(
+        "sycl2020_submodel",
+        values=("usm", "acc"),
+        when="+sycl2020",
+        default="usm",
+        description="SYCL2020 -> choose between usm and acc methods",
+    )
+    variant(
+        "std_submodel",
+        values=("data", "indices", "ranges"),
+        when="+std",
+        default="data",
+        description="STD -> choose between data, indices and ranges models",
+    )
+    variant(
+        "sycl_opencl_path",
+        values=str,
+        when="+sycl",
+        default="none",
+        description="[ComputeCpp only] Path to OpenCL library, usually called libOpenCL.so",
+    )
```

Comment:
This is essentially an external but it's modeled as a variant -- it should be a dependency on the `opencl` virtual so that the user can choose different opencl implementations.  Look at some of the OpenCL providers:

```console
> spack providers opencl
opencl:
cuda@:6  ocl-icd@2.2.3:2.2.7   ocl-icd@2.2.12   opencl-icd-loader  pocl@3:
cuda@7:  ocl-icd@2.2.8:2.2.11  ocl-icd@2.2.13:  pocl
```
and dependents:
```console
> spack dependents opencl
arrayfire  clinfo                      gdal        kassiopeia  opencv
bohrium    cmdstan                     gearshifft  lammps      py-pyopencl
clblast    dbcsr                       grass       libbeagle   sgpp
clfft      double-batched-fft-library  hpx5        onednn      specfem3d-globe
```

Reply:
Can probably just delete as ComputeCpp is gone